R/dvhydrograph-render.R

```diff
@@ -3,199 +3,291 @@ dvhydrographPlot <- function(data) {
+  yLabel <- paste0(refData[[series]][['type']], ", ", refData[[series]][['units']])
+  refData <- append(refData, parseDVApprovals(refData[[series]], timezone))
+
+  #Do Plotting
+  plot_object <- gsplot(ylog = logAxis, yaxs = 'i') %>%
+    grid(nx = NA, ny = NULL, lty = 3, col = "gray") %>%
+    axis(2, reverse = invertedFlag, las=0) %>%
+    view(xlim = c(startDate, endDate)) %>%
+    title(main = paste(ref_name_capital, "Reference Time Series")) %>%
+    legend(location = "below", cex = 0.8, y.intersp = 1.5)
+  
+  plot_object <-
+    XAxisLabelStyle(plot_object, startDate, endDate, timezone, plotDates)
+  
+  # for non-approval-bar objects
+  for (i in grep("^appr_", names(refData), invert = TRUE)) {
+    if(grepl("TimeSeries", names(refData[i]))){
+      refData[i][[names(refData[i])]] <- formatTimeSeriesForPlotting(refData[i][[names(refData[i])]], excludeZeroNegativeFlag)
+    }
+
+    refConfig <- getDVHydrographRefPlotConfig(refData[i], yLabel=yLabel)
+    for (j in seq_len(length(refConfig))) {
+      plot_object <- do.call(names(refConfig[j]), append(list(object = plot_object), refConfig[[j]]))
     }
-    
-    plot_object <- ApplyApprovalBarStyles(plot_object, refData)
-    
-    plot_object <- rmDuplicateLegendItems(plot_object)
-    
-    plot_object <- plot_object %>% 
-      abline(v=seq(from=startDate, to=endDate, by="days"), lty=3, col="gray", where='first') %>%
-      abline(v=seq(from=startDate, to=endDate, by="weeks"), col="darkgray", lwd=1, where='first')
-    
-    # patch up top extent of y-axis
-    plot_object <- RescaleYTop(plot_object)
-    
-    return(plot_object)
   }
+  
+  plot_object <- ApplyApprovalBarStyles(plot_object, refData)
+  
+  plot_object <- rmDuplicateLegendItems(plot_object)
+  
+  plot_object <- plot_object %>% 
+    abline(v=seq(from=startDate, to=endDate, by="days"), lty=3, col="gray", where='first') %>%
+    abline(v=seq(from=startDate, to=endDate, by="weeks"), col="darkgray", lwd=1, where='first')
+  
+  # patch up top extent of y-axis
+  plot_object <- RescaleYTop(plot_object)
+  
+  return(plot_object)
 }
 
-getDVHydrographPlotConfig <- function(reportObject, info = NULL, ...){
+#' Get DV Hydrograph Plot Config
+#'
+#' @description Given an item to plot, fetch the associated
+#' plot feature(s) and their styles. 
+#' @param plotItem the item to fetch styles and plot features for
+#' @param ... any additional parameters to pass into the function
+getDVHydrographPlotConfig <- function(plotItem, ...){
   styles <- getDvHydrographStyles()
 
-  x <- reportObject[[1]]$time
-  y <- reportObject[[1]]$value
-  legend.name <- reportObject[[1]]$legend.name
+  x <- plotItem[[1]]$time
+  y <- plotItem[[1]]$value
+  legend.name <- plotItem[[1]]$legend.name
   args <- list(...)
   
-  styles <- switch(names(reportObject), 
-    stat1Timeseries = list(
-      lines = append(list(x=x, y=y, legend.name=legend.name), styles$stat1_lines)
+  styles <- switch(names(plotItem), 
```

Comment:
I think we should add an exception after this switch statement if nothing is found. Would cover the case where someone is asking for a plotItem that's not supported/expected.

I will do that for UV Hydrographs.